web/modules/custom/joinup_federation/src/Plugin/Adms2ConvertPass/Pass45.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_federation\Plugin\Adms2ConvertPass;
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\joinup_federation\JoinupFederationAdms2ConvertPassPluginBase;
+use Drupal\Tests\joinup_federation\Kernel\ConvertToAdms2StepTest;
+
+/**
+ * Conversion Pass #45.
+ *
+ * URI: dct:publisher
+ * Type: N.A.
+ * Action: Updated
+ * Description:
+ * - Updated: Cardinality: 1..n -> 1..1.
+ * - Updated the definition: the publisher is the Agent that publishes the asset
+ *   or solutions, not the Agent that publishes the metadata about it.
+ *
+ * @Adms2ConvertPass(
+ *   id = "pass_45",
+ * )
+ */
+class Pass45 extends JoinupFederationAdms2ConvertPassPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function convert(array $data): void {
+    $results = $this->getTriplesFromGraph($data['sink_graph'],
+      NULL,
+      'http://www.w3.org/ns/dcat#publisher'
+    );
+
+    $entities = [];
+    // In order to be fair, and since the notion of deltas does not exist in
+    // SPARQL, we are only going to keep the first instance found for the given
+    // value.
+    foreach ($results as $result) {
+      if (isset($entities[$result['subject']])) {
+        continue;
+      }
+      $entities[$result['subject']] = $result['object'];
+    }
+
+    foreach ($entities as $subject => $object) {
+      $query = <<<QUERY
+WITH <{$data['sink_graph']}>
+DELETE { <$subject> <http://www.w3.org/ns/dcat#publisher> ?publisher }
+INSERT { <$subject> <http://www.w3.org/ns/dcat#publisher> <$object> }
```

Comment:
I'm a bit conflicted here, but as we can always re-federate, it is probably fine to discard the other publishers.

Reply:
This is also going to be removed. Sorry for not having done that yet.